internal/certdb/certdb.go

```diff
@@ -0,0 +1,115 @@
+package certdb
+
+import (
+	"database/sql"
+	"fmt"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const queryCreateTable = "CREATE TABLE IF NOT EXISTS %s (CSR VARCHAR PRIMARY KEY UNIQUE NOT NULL, Certificate VARCHAR)"
+
+const queryGetAllCSRs = "SELECT rowid, * FROM %s"
+const queryGetCSR = "SELECT rowid, * FROM %s WHERE CSR=?"
+const queryCreateCSR = "INSERT INTO %s (CSR) VALUES (?)"
+const queryUpdateCSR = "UPDATE %s SET Certificate=? WHERE CSR=?"
+const queryDeleteCSR = "DELETE FROM %s WHERE CSR=?"
+
+type CertificateRequests struct {
+	table string
+	conn  *sql.DB
+}
+
+type CertificateRequest struct {
+	ID          int
+	CSR         string
+	Certificate *string
+}
+
+func (db *CertificateRequests) RetrieveAll() ([]*CertificateRequest, error) {
```

Comment:
I understand the reason for using a pointer receiver for this function, but not sure about the return being a pointer. The struct only contains basic types so should be relatively cheap to copy.

Reply:
The same comment apply for all the return values that are pointers.

Reply:
I guess I'm used to using pointers by default. Updated.